openlibrary/catalog/add_book/match.py

```diff
@@ -53,13 +47,434 @@ def editions_match(candidate, existing):
     ):
         if existing.get(f):
             rec2[f] = existing[f]
+    # Transfer authors as Dicts str: str
     if existing.authors:
         rec2['authors'] = []
-        for a in existing.authors:
-            while a.type.key == '/type/redirect':
-                a = web.ctx.site.get(a.location)
-            if a.type.key == '/type/author':
-                assert a['name']
-                rec2['authors'].append({'name': a['name'], 'db_name': db_name(a)})
+    for a in existing.authors:
+        while a.type.key == '/type/redirect':
+            a = web.ctx.site.get(a.location)
+        if a.type.key == '/type/author':
+            author = {'name': a['name']}
+            if birth := a.get('birth_date'):
+                author['birth_date'] = birth
+            if death := a.get('death_date'):
+                author['death_date'] = death
+            rec2['authors'].append(author)
+    return threshold_match(rec, rec2, threshold)
+
+
+def normalize(s: str) -> str:
+    """
+    Normalizes a title for matching purposes, not display,
+    by lowercasing, unicode -> NFC,
+    stripping extra whitespace and punctuation, and replacing ampersands.
+    """
+
+    if isinstance(s, str):
+        # LATIN SMALL LETTER L WITH STROKE' (U+0142) -> 'l'
+        s = unicodedata.normalize('NFC', s.replace('\u0142', 'l'))
```

Comment:
This is awfully specific fixing L WITH STROKE, can it not be done along with all other similar normalizations?  and what else could s be other than a str here?

Reply:
I agree that this should move to a table of replacement values (even if it starts off as a table of 1 entry). The `isinstance(s, str)` could be left over from a Python 2->3 update.

Some other potential entries for the table are here: https://stackoverflow.com/questions/1410308/how-to-implement-unicode-string-matching-by-folding-in-python

Another approach is to use collation keys of an appropriate "strength" using `locale.strcoll`/`locale.strxfrm` or [PyICU](https://pypi.org/project/PyICU/), although they focus more on "correct" as opposed to "commonly used" transforms.